proton

```diff
@@ -1599,6 +1601,48 @@ class Session:
         self.log_file = open(lfile_path, "a")
         return True
 
+    def setup_local_shader_cache(self):
+        path = os.environ.get("STEAM_COMPAT_SHADER_PATH", "")
+        if not path:
+            return
+        shader_cache_name = "steamapp_shader_cache"
+        shader_cache_vars = {
+            # Nvidia
+            "__GL_SHADER_DISK_CACHE_APP_NAME": shader_cache_name,
+            "__GL_SHADER_DISK_CACHE_PATH": os.path.join(path, "nvidiav1"),
+            "__GL_SHADER_DISK_CACHE_READ_ONLY_APP_NAME": "steam_shader_cache;steamapp_merged_shader_cache",
+            "__GL_SHADER_DISK_CACHE_SIZE": "5000000000",
+            #"__GL_SHADER_DISK_CACHE_SKIP_CLEANUP": "1",
+            # Mesa
+            "MESA_DISK_CACHE_READ_ONLY_FOZ_DBS": "steam_cache,steam_precompiled",
+            "MESA_DISK_CACHE_SINGLE_FILE": "1",
+            "MESA_GLSL_CACHE_DIR": path,
+            "MESA_GLSL_CACHE_MAX_SIZE": "5G",
+            "MESA_SHADER_CACHE_DIR": path,
+            "MESA_SHADER_CACHE_MAX_SIZE": "5G",
+            # AMD VK
+            "AMD_VK_PIPELINE_CACHE_FILENAME": shader_cache_name,
+            "AMD_VK_PIPELINE_CACHE_PATH": os.path.join(path, "AMDv1"),
+            "AMD_VK_USE_PIPELINE_CACHE": "1",
+            # DXVK
+            "DXVK_STATE_CACHE_PATH": os.path.join(path, "DXVK_state_cache"),
```

Comment:
dxvk state cache no longer exists

Reply:
Indeed but Steam still sets it, so kept it for parity.